dev/devicelab/lib/framework/runner.dart

```diff
@@ -24,19 +25,28 @@ Future<Map<String, dynamic>> runTask(
   bool silent = false,
   String localEngine,
   String localEngineSrcPath,
+  String deviceId,
 }) async {
   final String taskExecutable = 'bin/tasks/$taskName.dart';
 
   if (!file(taskExecutable).existsSync())
     throw 'Executable Dart file not found: $taskExecutable';
 
-  final Process runner = await startProcess(dartBin, <String>[
-    '--enable-vm-service=0', // zero causes the system to choose a free port
-    '--no-pause-isolates-on-exit',
-    if (localEngine != null) '-DlocalEngine=$localEngine',
-    if (localEngineSrcPath != null) '-DlocalEngineSrcPath=$localEngineSrcPath',
-    taskExecutable,
-  ]);
+  Map<String, String> environment;
+  if (deviceId != null) {
+    environment ??= <String, String>{DeviceIdEnvName: deviceId};
```

Comment:
`environment `  is always null here, just use `=`.

Is there any reason you are conditionally passing environment? You could instead conditionally insert the device id:

```
environment = <String, String>{
  if (deviceId != null)
    DeviceIdEnvName: deviceId
}
```

Reply:
I was thinking maybe in the future there will be more environment variables adding. But yeah I should have made it simpler.